galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -380,7 +383,8 @@ def get(self, request, *args, **kwargs):
         if content_bind and content_bind.startswith("unix:"):
             response = self._get_unix_socket_response(url)
         else:
-            response = self._get_tcp_response(url)
+            distribution = self._get_ansible_distribution(self.kwargs['path'])
```

Comment:
Right now we’re proxying the requests for the content app for on prem on cloud. We still have to proxy the requests to `X_PULP_CONTENT_HOST` because console.redhat.com doesn’t expose the content app, but we need to stop proxying the requests when running on premise.

I think the best way to fix this is to check if the app is running in insights mode. If it is, proxy the request to `X_PULP_CONTENT_HOST` like we do now. If it's not, generate a URL for `{CONTENT_ORIGIN}/{CONTENT_PATH_PREFIX}`, pre authenticate it and return a redirect to it.

You can get rid of `_get_tcp_response` and `_get_unix_socket_response` since those were used for proxying on prem, and aren't needed anymore.